### DISCUSSION TITLE
Is there a simple way to fade a scene?

### DISCUSSION THREAD

**Discussion by vascanera** [replit/kaboom#233] [General]
Let's say I have this intro scene that has some nice animations in it (looping cloud movement, birds flying, etc.).
When the user taps / clicks the Play button, what would be a quick and clean way of fading out the whole intro scene and then fading in the next scene?
Thanks!

**Comment by oaklandgit** [comment]
@vascanera Kaboom has no "root object" that you could fade out as I understand. But I wonder if you could cheat this a bit. Instead of fading out everything, add a white rectangle (or whatever color your background is) on top of everything and fade it _in_. I may try this in the next few days. I'll post here if I get that working or discover a better way.

**Comment by triptych** [comment]
This is on a web page right? You could use CSS to slowly change the opacity of the <canvas> element.

**Reply by vascanera** [reply]
You are joking, right? 
I am talking about fading INSIDE the canvas (or webgl surface). 
Maybe, keep the UI intact, and just fade the non-UI elements.

**Reply by triptych** [reply]
I thought you wanted to fade the whole viewport and was just offering another alternative. The idea of having an element the size of the viewport and slowly fading it in would probably be worth trying.

**Comment by slmjkdbtl** [comment]
It's planned to let you program scene transition, but don't have a design yet. 

Some options to configure the transition will make sense in `go()`, but currently everything you pass in `go(scene, ...)` after the scene name are arguments that gets passed to the scene initializer, don't have an idea on what's a good way to build scene transition interface yet, except like what @oaklandgit said to use a rect as filter together with stuff like `wait()` and `opacity`. I can also make an example for that.

Any proposal is welcome
